packages/fast-components-react-base/src/viewport-positioner/viewport-positioner.tsx

```diff
@@ -0,0 +1,1011 @@
+        }
+
+        if (this.props.verticalPositioningMode !== AxisPositioningMode.uncontrolled) {
+            const verticalOptions: ViewportPositionerVerticalPositionLabel[] = this.getVerticalPositioningOptions();
+            desiredVerticalPosition = this.state.defaultVerticalPosition;
+
+            const verticalThreshold: number =
+                this.props.verticalThreshold !== undefined
+                    ? this.props.verticalThreshold
+                    : this.positionerRect.height;
+
+            if (
+                desiredVerticalPosition ===
+                    ViewportPositionerVerticalPositionLabel.undefined ||
+                this.getOptionHeight(desiredVerticalPosition) < verticalThreshold
+            ) {
+                desiredVerticalPosition =
+                    this.getOptionHeight(verticalOptions[0]) >
+                    this.getOptionHeight(verticalOptions[1])
+                        ? verticalOptions[0]
+                        : verticalOptions[1];
+            }
+        }
+
+        this.setState(Object.assign(
+            {
+                xTranslate: this.calculateHorizontalTranslate(desiredHorizontalPosition),
+                yTranslate: this.calculateVerticalTranslate(desiredVerticalPosition),
+                initialLayoutComplete: true,
+            },
+            this.getBaseHorizontalPositioningState(desiredHorizontalPosition),
+            this.getBaseVerticalPositioningState(desiredVerticalPosition)
+        ) as ViewportPositionerState);
+    };
+
+    /**
+     * Get base horizontal positioning state based on desired position
+     */
+    private getBaseHorizontalPositioningState = (
+        desiredHorizontalPosition: ViewportPositionerHorizontalPositionLabel
+    ): Partial<ViewportPositionerState> => {
+        let right: number = null;
+        let left: number = null;
+        let xTransformOrigin: string = "left";
+
+        switch (desiredHorizontalPosition) {
+            case ViewportPositionerHorizontalPositionLabel.left:
+                xTransformOrigin = "right";
+                right = this.positionerRect.width;
+                break;
+
+            case ViewportPositionerHorizontalPositionLabel.insetLeft:
+                xTransformOrigin = "right";
+                right = 0;
+                break;
+
+            case ViewportPositionerHorizontalPositionLabel.insetRight:
+                xTransformOrigin = "left";
+                left = 0;
+                break;
+
+            case ViewportPositionerHorizontalPositionLabel.right:
+                xTransformOrigin = "left";
+                left = this.anchorWidth;
+                break;
+        }
+
+        return {
+            xTransformOrigin,
+            right,
+            left,
+            currentHorizontalPosition: desiredHorizontalPosition,
+        };
+    };
+
+    /**
+     * Get base vertical positioning state based on desired position
+     */
+    private getBaseVerticalPositioningState = (
```

Comment:
"getDesiredVerticalPositioningState"

Reply:
Actually I think a simple "geVerticalPositioningState" would be fine